storage/src/test/java/tech/pegasys/teku/storage/server/DepositStorageTest.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.storage.server;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.common.eventbus.EventBus;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import tech.pegasys.teku.datastructures.util.DataStructureUtil;
+import tech.pegasys.teku.pow.api.Eth1EventsChannel;
+import tech.pegasys.teku.pow.event.DepositsFromBlockEvent;
+import tech.pegasys.teku.pow.event.MinGenesisTimeBlockEvent;
+import tech.pegasys.teku.storage.server.rocksdb.AbstractRocksDbDatabaseTest;
+import tech.pegasys.teku.storage.server.rocksdb.RocksDbConfiguration;
+import tech.pegasys.teku.storage.server.rocksdb.RocksDbDatabase;
+import tech.pegasys.teku.util.async.SafeFuture;
+import tech.pegasys.teku.util.config.StateStorageMode;
+
+public class DepositStorageTest extends AbstractRocksDbDatabaseTest {
+  private final EventBus eventBus = new EventBus();
+  private final TrackingEth1EventsChannel eventsChannel = new TrackingEth1EventsChannel();
+  final DataStructureUtil dataStructureUtil = new DataStructureUtil();
+  private DepositStorage depositStorage;
+
+  @BeforeEach
+  public void beforeEach() {
+    depositStorage = DepositStorage.create(eventBus, eventsChannel, database);
+    depositStorage.start();
+  }
+
+  @Test
+  public void shouldSendGenesisBeforeFirstDeposit()
+      throws ExecutionException, InterruptedException {
+    DepositsFromBlockEvent blockEvent = dataStructureUtil.randomDepositsFromBlockEvent(100L, 10);
+    MinGenesisTimeBlockEvent genesis = dataStructureUtil.randomMinGenesisTimeBlockEvent(1L);
+    database.addDepositsFromBlockEvent(blockEvent);
+    database.addMinGenesisTimeBlock(genesis);
```

Comment:
nit: the tests will be slightly easier to read if we add deposits and min genesis events in the order they actually happen.  That also means the test follows the sequence requirements that apply in the real code as well (won't affect the result but good to be accurate).